<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Kenong Su @Emory, Zhijin Wu @Brown, Hao Wu @Emory" />


<title>POWSC: Power and Sample Size Analysis for single-cell RNA-seq</title>

<script>// Hide empty <a> tag within highlighted CodeBlock for screen reader accessibility (see https://github.com/jgm/pandoc/issues/6352#issuecomment-626106786) -->
// v0.0.1
// Written by JooYoung Seo (jooyoung@psu.edu) and Atsushi Yasumoto on June 1st, 2020.

document.addEventListener('DOMContentLoaded', function() {
  const codeList = document.getElementsByClassName("sourceCode");
  for (var i = 0; i < codeList.length; i++) {
    var linkList = codeList[i].getElementsByTagName('a');
    for (var j = 0; j < linkList.length; j++) {
      if (linkList[j].innerHTML === "") {
        linkList[j].setAttribute('aria-hidden', 'true');
      }
    }
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore"><strong><em>POWSC: Power and Sample Size Analysis for single-cell RNA-seq</em></strong></h1>
<h4 class="author">Kenong Su <span class="citation">@Emory</span>, Zhijin Wu <span class="citation">@Brown</span>, Hao Wu <span class="citation">@Emory</span></h4>





<div id="abstract" class="section level2">
<h2>1. Abstract</h2>
<p>Determining the sample size for adequate power to detect statistical significance is a crucial step at the design stage for high-throughput experiments. Even though a number of methods and tools are available for sample size calculation for microarray and RNA-seq under the context of differential expression, this topic in the field of single-cell RNA sequencing is understudied. Moreover, the unique data characteristics present in scRNA-seq including sparsity and heterogeneity gain the challenge.</p>
</div>
<div id="introduction" class="section level2">
<h2>2. Introduction</h2>
<p><code>POWSC</code> is an R package designed for power assessment and sample size estimation in scRNA-seq experiment. It contains three main functionalities: (1). <strong>Parameter Estimation</strong>: adopted and modified from the core of <a href="https://github.com/haowulab/SC2P">SC2P</a>. (2). <strong>Data Simulation</strong>: consider two cases: paired-wise comparison and multiple comparisons. (3). <strong>Power Evaluation</strong>: provide both stratified (detailed) and marginal powers</p>
</div>
<div id="use-powsc" class="section level2">
<h2>3. Use POWSC</h2>
<p>In the context of differential expression (DE) analysis, scientists are usually interested in two different scenarios: (1) within cell type: comparing the same cell types across biological conditions such as case vs.Â control, which reveals the expression change of a particular cell type under different contexts. (2) between cell types: comparing different cell types under the same condition, which identifies biomarkers to distinguish cell types. In either case, the experiment starts from a number of cells randomly picked from a tissue sample consisting of a mixture of different cell types. The only factor one can control is the total number of cells.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">suppressMessages</span>(<span class="kw">library</span>(POWSC))</span></code></pre></div>
<div id="for-two-group-comparison" class="section level4">
<h4>(1). For two-group comparison</h4>
<p>In the first scenario, the numbers of cells for a particular cell type in different biological conditions are often similar, barring significant changes in cell composition. It uses one cell type as the benchmark data and perturbs the genes with mixture proportion (<span class="math inline">\(\pi\)</span>) and log fold changes (<span class="math inline">\(lfc\)</span>) as the DE genes according two forms.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw">data</span>(<span class="st">&quot;es_mef_sce&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a>sce =<span class="st"> </span>es_mef_sce[, <span class="kw">colData</span>(es_mef_sce)<span class="op">$</span>cellTypes <span class="op">==</span><span class="st"> &quot;fibro&quot;</span>]</span>
<span id="cb2-3"><a href="#cb2-3"></a>estParas =<span class="st"> </span><span class="kw">Est2Phase</span>(sce)</span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="co"># Users can customize how many cells they need as to change the number of n. </span></span>
<span id="cb2-5"><a href="#cb2-5"></a>simData =<span class="st"> </span><span class="kw">Simulate2SCE</span>(<span class="dt">n=</span><span class="dv">100</span>, <span class="dt">estParas1 =</span> estParas, <span class="dt">estParas2 =</span> estParas)</span>
<span id="cb2-6"><a href="#cb2-6"></a>de =<span class="st"> </span><span class="kw">runMAST</span>(simData<span class="op">$</span>sce)</span>
<span id="cb2-7"><a href="#cb2-7"></a>estPower1 =<span class="st"> </span><span class="kw">Power_Disc</span>(de, <span class="dt">simData =</span> simData)</span>
<span id="cb2-8"><a href="#cb2-8"></a>estPower2 =<span class="st"> </span><span class="kw">Power_Cont</span>(de, <span class="dt">simData =</span> simData)</span></code></pre></div>
</div>
<div id="for-multiple-group-comparisons" class="section level4">
<h4>(2). For multiple-group comparisons</h4>
<p>In the second scenario, the numbers of cells for distinct cell types can be very different, so the power for DE highly depends on the mixing proportions. The sample data GSE67835 can be download here: <a href="https://www.dropbox.com/s/55zdktqfqiwfs3l/GSE67835.RData?dl=0" class="uri">https://www.dropbox.com/s/55zdktqfqiwfs3l/GSE67835.RData?dl=0</a>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>sim_size =<span class="st"> </span><span class="dv">1000</span></span>
<span id="cb3-2"><a href="#cb3-2"></a>cell_per =<span class="st"> </span><span class="kw">c</span>(<span class="fl">0.2</span>, <span class="fl">0.3</span>, <span class="fl">0.5</span>)</span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="kw">load</span>(<span class="st">&quot;pathto/GSE67835.RData&quot;</span>)</span>
<span id="cb3-4"><a href="#cb3-4"></a>col =<span class="st"> </span><span class="kw">colData</span>(sce)</span>
<span id="cb3-5"><a href="#cb3-5"></a>exprs =<span class="st"> </span><span class="kw">assays</span>(sce)<span class="op">$</span>counts</span>
<span id="cb3-6"><a href="#cb3-6"></a>(<span class="dt">tb =</span> <span class="kw">table</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>Patients, <span class="kw">colData</span>(sce)<span class="op">$</span>cellTypes))</span>
<span id="cb3-7"><a href="#cb3-7"></a><span class="co"># use AB_S7 patient as example and take three cell types: astrocytes hybrid and neurons</span></span>
<span id="cb3-8"><a href="#cb3-8"></a>estParas_set =<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb3-9"><a href="#cb3-9"></a>celltypes =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;oligodendrocytes&quot;</span>, <span class="st">&quot;hybrid&quot;</span>, <span class="st">&quot;neurons&quot;</span>)</span>
<span id="cb3-10"><a href="#cb3-10"></a><span class="cf">for</span> (cp <span class="cf">in</span> celltypes){</span>
<span id="cb3-11"><a href="#cb3-11"></a>    <span class="kw">print</span>(cp)</span>
<span id="cb3-12"><a href="#cb3-12"></a>    ix =<span class="st"> </span><span class="kw">intersect</span>(<span class="kw">grep</span>(cp, col<span class="op">$</span>cellTypes), <span class="kw">grep</span>(<span class="st">&quot;AB_S7&quot;</span>, col<span class="op">$</span>Patients))</span>
<span id="cb3-13"><a href="#cb3-13"></a>    tmp_mat =<span class="st"> </span>exprs[, ix]</span>
<span id="cb3-14"><a href="#cb3-14"></a>    tmp_paras =<span class="st"> </span><span class="kw">Est2Phase</span>(tmp_mat)</span>
<span id="cb3-15"><a href="#cb3-15"></a>    estParas_set[[cp]] =<span class="st"> </span>tmp_paras</span>
<span id="cb3-16"><a href="#cb3-16"></a>}</span>
<span id="cb3-17"><a href="#cb3-17"></a><span class="co">######### </span></span>
<span id="cb3-18"><a href="#cb3-18"></a><span class="co">#########  Simulation part</span></span>
<span id="cb3-19"><a href="#cb3-19"></a><span class="co">######### </span></span>
<span id="cb3-20"><a href="#cb3-20"></a>sim =<span class="st"> </span><span class="kw">SimulateMultiSCEs</span>(<span class="dt">n =</span> <span class="dv">6000</span>, <span class="dt">estParas_set =</span> estParas_set, <span class="dt">multiProb =</span> cell_per)</span>
<span id="cb3-21"><a href="#cb3-21"></a></span>
<span id="cb3-22"><a href="#cb3-22"></a><span class="co">######### </span></span>
<span id="cb3-23"><a href="#cb3-23"></a><span class="co">#########  DE analysis part</span></span>
<span id="cb3-24"><a href="#cb3-24"></a><span class="co">######### </span></span>
<span id="cb3-25"><a href="#cb3-25"></a>DE_rslt =<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb3-26"><a href="#cb3-26"></a><span class="cf">for</span> (comp <span class="cf">in</span> <span class="kw">names</span>(sim)){</span>
<span id="cb3-27"><a href="#cb3-27"></a>    tmp =<span class="st"> </span><span class="kw">runMAST</span>(sim[[comp]]<span class="op">$</span>sce)</span>
<span id="cb3-28"><a href="#cb3-28"></a>    DE_rslt[[comp]] =<span class="st"> </span>tmp</span>
<span id="cb3-29"><a href="#cb3-29"></a>}</span>
<span id="cb3-30"><a href="#cb3-30"></a></span>
<span id="cb3-31"><a href="#cb3-31"></a><span class="co">######### </span></span>
<span id="cb3-32"><a href="#cb3-32"></a><span class="co">######### Summarize the power result</span></span>
<span id="cb3-33"><a href="#cb3-33"></a><span class="co">######### </span></span>
<span id="cb3-34"><a href="#cb3-34"></a>pow_rslt =<span class="st"> </span>pow1 =<span class="st"> </span>pow2 =<span class="st"> </span>pow1_marg =<span class="st"> </span>pow2_marg =<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb3-35"><a href="#cb3-35"></a>TD =<span class="st"> </span>CD =<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb3-36"><a href="#cb3-36"></a><span class="cf">for</span> (comp <span class="cf">in</span> <span class="kw">names</span>(sim)){</span>
<span id="cb3-37"><a href="#cb3-37"></a>    tmp1 =<span class="st"> </span><span class="kw">Power_Disc</span>(DE_rslt[[comp]], sim[[comp]])</span>
<span id="cb3-38"><a href="#cb3-38"></a>    tmp2 =<span class="st"> </span><span class="kw">Power_Cont</span>(DE_rslt[[comp]], sim[[comp]])</span>
<span id="cb3-39"><a href="#cb3-39"></a>    TD =<span class="st"> </span><span class="kw">c</span>(TD, tmp2<span class="op">$</span>TD); CD =<span class="st"> </span><span class="kw">c</span>(CD, tmp2<span class="op">$</span>CD)</span>
<span id="cb3-40"><a href="#cb3-40"></a>    pow1_marg =<span class="st"> </span><span class="kw">c</span>(pow1_marg, tmp1<span class="op">$</span>power.marginal)</span>
<span id="cb3-41"><a href="#cb3-41"></a>    pow2_marg =<span class="st"> </span><span class="kw">c</span>(pow2_marg, tmp2<span class="op">$</span>power.marginal)</span>
<span id="cb3-42"><a href="#cb3-42"></a>    pow_rslt[[comp]] =<span class="st"> </span><span class="kw">list</span>(<span class="dt">pow1 =</span> tmp1, <span class="dt">pow2 =</span> tmp2)</span>
<span id="cb3-43"><a href="#cb3-43"></a>    pow1 =<span class="st"> </span><span class="kw">rbind</span>(pow1, tmp1<span class="op">$</span>power)</span>
<span id="cb3-44"><a href="#cb3-44"></a>    pow2 =<span class="st"> </span><span class="kw">rbind</span>(pow2, tmp2<span class="op">$</span>power)</span>
<span id="cb3-45"><a href="#cb3-45"></a>}</span>
<span id="cb3-46"><a href="#cb3-46"></a></span>
<span id="cb3-47"><a href="#cb3-47"></a><span class="co">######### </span></span>
<span id="cb3-48"><a href="#cb3-48"></a><span class="co">######### Demonstrate the result by heatmap</span></span>
<span id="cb3-49"><a href="#cb3-49"></a><span class="co">######### </span></span>
<span id="cb3-50"><a href="#cb3-50"></a><span class="kw">library</span>(RColorBrewer); <span class="kw">library</span>(pheatmap)</span>
<span id="cb3-51"><a href="#cb3-51"></a>breaksList =<span class="st"> </span><span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">by =</span> <span class="fl">0.01</span>)</span>
<span id="cb3-52"><a href="#cb3-52"></a>colors =<span class="st"> </span><span class="kw">colorRampPalette</span>(<span class="kw">rev</span>(<span class="kw">brewer.pal</span>(<span class="dt">n =</span> <span class="dv">7</span>, <span class="dt">name =</span> <span class="st">&quot;RdYlBu&quot;</span>)))(<span class="kw">length</span>(breaksList))</span>
<span id="cb3-53"><a href="#cb3-53"></a><span class="kw">dimnames</span>(pow1) =<span class="st"> </span><span class="kw">list</span>(<span class="kw">names</span>(sim), <span class="kw">names</span>(tmp1<span class="op">$</span>CD))</span>
<span id="cb3-54"><a href="#cb3-54"></a><span class="kw">dimnames</span>(pow2) =<span class="st"> </span><span class="kw">list</span>(<span class="kw">names</span>(sim), <span class="kw">names</span>(tmp2<span class="op">$</span>CD))</span>
<span id="cb3-55"><a href="#cb3-55"></a><span class="kw">pheatmap</span>(pow1, <span class="dt">display_numbers =</span> T, <span class="dt">color=</span>colors, <span class="dt">show_rownames =</span> T,</span>
<span id="cb3-56"><a href="#cb3-56"></a>         <span class="dt">cellwidth =</span> <span class="dv">50</span>, <span class="dt">cellheight =</span> <span class="dv">50</span>, <span class="dt">legend =</span> T,</span>
<span id="cb3-57"><a href="#cb3-57"></a>         <span class="dt">border_color =</span> <span class="st">&quot;grey96&quot;</span>, <span class="dt">na_col =</span> <span class="st">&quot;grey&quot;</span>,</span>
<span id="cb3-58"><a href="#cb3-58"></a>         <span class="dt">cluster_row =</span> <span class="ot">FALSE</span>, <span class="dt">cluster_cols =</span> <span class="ot">FALSE</span>,</span>
<span id="cb3-59"><a href="#cb3-59"></a>         <span class="dt">breaks =</span> <span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">0.01</span>),</span>
<span id="cb3-60"><a href="#cb3-60"></a>         <span class="dt">main =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="session-info" class="section level2">
<h2>4. Session Info</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="kw">sessionInfo</span>()</span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="co">#&gt; R version 4.0.3 (2020-10-10)</span></span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="co">#&gt; Running under: macOS Big Sur 10.16</span></span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="co">#&gt; </span></span>
<span id="cb4-6"><a href="#cb4-6"></a><span class="co">#&gt; Matrix products: default</span></span>
<span id="cb4-7"><a href="#cb4-7"></a><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span></span>
<span id="cb4-8"><a href="#cb4-8"></a><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span></span>
<span id="cb4-9"><a href="#cb4-9"></a><span class="co">#&gt; </span></span>
<span id="cb4-10"><a href="#cb4-10"></a><span class="co">#&gt; locale:</span></span>
<span id="cb4-11"><a href="#cb4-11"></a><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span id="cb4-12"><a href="#cb4-12"></a><span class="co">#&gt; </span></span>
<span id="cb4-13"><a href="#cb4-13"></a><span class="co">#&gt; attached base packages:</span></span>
<span id="cb4-14"><a href="#cb4-14"></a><span class="co">#&gt; [1] stats4    parallel  stats     graphics  grDevices utils     datasets </span></span>
<span id="cb4-15"><a href="#cb4-15"></a><span class="co">#&gt; [8] methods   base     </span></span>
<span id="cb4-16"><a href="#cb4-16"></a><span class="co">#&gt; </span></span>
<span id="cb4-17"><a href="#cb4-17"></a><span class="co">#&gt; other attached packages:</span></span>
<span id="cb4-18"><a href="#cb4-18"></a><span class="co">#&gt;  [1] POWSC_0.99.0                MAST_1.16.0                </span></span>
<span id="cb4-19"><a href="#cb4-19"></a><span class="co">#&gt;  [3] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0</span></span>
<span id="cb4-20"><a href="#cb4-20"></a><span class="co">#&gt;  [5] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        </span></span>
<span id="cb4-21"><a href="#cb4-21"></a><span class="co">#&gt;  [7] IRanges_2.24.1              S4Vectors_0.28.1           </span></span>
<span id="cb4-22"><a href="#cb4-22"></a><span class="co">#&gt;  [9] MatrixGenerics_1.2.1        matrixStats_0.58.0         </span></span>
<span id="cb4-23"><a href="#cb4-23"></a><span class="co">#&gt; [11] limma_3.46.0                Biobase_2.50.0             </span></span>
<span id="cb4-24"><a href="#cb4-24"></a><span class="co">#&gt; [13] BiocGenerics_0.36.0        </span></span>
<span id="cb4-25"><a href="#cb4-25"></a><span class="co">#&gt; </span></span>
<span id="cb4-26"><a href="#cb4-26"></a><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span id="cb4-27"><a href="#cb4-27"></a><span class="co">#&gt;  [1] sass_0.3.1             pkgload_1.2.0          jsonlite_1.7.2        </span></span>
<span id="cb4-28"><a href="#cb4-28"></a><span class="co">#&gt;  [4] bslib_0.2.4            assertthat_0.2.1       GenomeInfoDbData_1.2.4</span></span>
<span id="cb4-29"><a href="#cb4-29"></a><span class="co">#&gt;  [7] yaml_2.2.1             remotes_2.2.0          sessioninfo_1.1.1     </span></span>
<span id="cb4-30"><a href="#cb4-30"></a><span class="co">#&gt; [10] pillar_1.5.0           lattice_0.20-41        glue_1.4.2            </span></span>
<span id="cb4-31"><a href="#cb4-31"></a><span class="co">#&gt; [13] digest_0.6.27          RColorBrewer_1.1-2     XVector_0.30.0        </span></span>
<span id="cb4-32"><a href="#cb4-32"></a><span class="co">#&gt; [16] colorspace_2.0-0       plyr_1.8.6             htmltools_0.5.1.1     </span></span>
<span id="cb4-33"><a href="#cb4-33"></a><span class="co">#&gt; [19] Matrix_1.3-2           pkgconfig_2.0.3        devtools_2.3.2        </span></span>
<span id="cb4-34"><a href="#cb4-34"></a><span class="co">#&gt; [22] pheatmap_1.0.12        zlibbioc_1.36.0        purrr_0.3.4           </span></span>
<span id="cb4-35"><a href="#cb4-35"></a><span class="co">#&gt; [25] scales_1.1.1           processx_3.4.5         tibble_3.1.0          </span></span>
<span id="cb4-36"><a href="#cb4-36"></a><span class="co">#&gt; [28] generics_0.1.0         ggplot2_3.3.3          usethis_2.0.1         </span></span>
<span id="cb4-37"><a href="#cb4-37"></a><span class="co">#&gt; [31] ellipsis_0.3.1         cachem_1.0.4           withr_2.4.1           </span></span>
<span id="cb4-38"><a href="#cb4-38"></a><span class="co">#&gt; [34] cli_2.3.1              magrittr_2.0.1         crayon_1.4.1          </span></span>
<span id="cb4-39"><a href="#cb4-39"></a><span class="co">#&gt; [37] memoise_2.0.0          evaluate_0.14          ps_1.6.0              </span></span>
<span id="cb4-40"><a href="#cb4-40"></a><span class="co">#&gt; [40] fs_1.5.0               fansi_0.4.2            pkgbuild_1.2.0        </span></span>
<span id="cb4-41"><a href="#cb4-41"></a><span class="co">#&gt; [43] tools_4.0.3            data.table_1.14.0      prettyunits_1.1.1     </span></span>
<span id="cb4-42"><a href="#cb4-42"></a><span class="co">#&gt; [46] lifecycle_1.0.0        stringr_1.4.0          munsell_0.5.0         </span></span>
<span id="cb4-43"><a href="#cb4-43"></a><span class="co">#&gt; [49] DelayedArray_0.16.2    callr_3.5.1            compiler_4.0.3        </span></span>
<span id="cb4-44"><a href="#cb4-44"></a><span class="co">#&gt; [52] jquerylib_0.1.3        rlang_0.4.10           grid_4.0.3            </span></span>
<span id="cb4-45"><a href="#cb4-45"></a><span class="co">#&gt; [55] RCurl_1.98-1.2         rstudioapi_0.13        bitops_1.0-6          </span></span>
<span id="cb4-46"><a href="#cb4-46"></a><span class="co">#&gt; [58] rmarkdown_2.7          testthat_3.0.2         gtable_0.3.0          </span></span>
<span id="cb4-47"><a href="#cb4-47"></a><span class="co">#&gt; [61] DBI_1.1.1              abind_1.4-5            reshape2_1.4.4        </span></span>
<span id="cb4-48"><a href="#cb4-48"></a><span class="co">#&gt; [64] R6_2.5.0               knitr_1.31             dplyr_1.0.4           </span></span>
<span id="cb4-49"><a href="#cb4-49"></a><span class="co">#&gt; [67] fastmap_1.1.0          utf8_1.1.4             rprojroot_2.0.2       </span></span>
<span id="cb4-50"><a href="#cb4-50"></a><span class="co">#&gt; [70] desc_1.2.0             stringi_1.5.3          Rcpp_1.0.6            </span></span>
<span id="cb4-51"><a href="#cb4-51"></a><span class="co">#&gt; [73] vctrs_0.3.6            tidyselect_1.1.0       xfun_0.21</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
